content/blogs/2024-03-11-kestra-airbyte.md

```diff
@@ -0,0 +1,139 @@
+
+In this post, we would be setting up Airbyte and Kestra on two different AWS EC2 instances, create a new connection in Airbyte that will capture data from RDS MySQL database and sync it to S3 bucket. We will then use Kestra to invoke the Sync operation, as well as check the status of the Sync job. We can then go to the S3 bucket and confirm that the Kestra tasks indeed interacted with Airbyte and performed the requested sync.
+
+### Setup Airbyte cluster ###
+
+In order to create Airbyte cluster on AWS EC2 machine, we will be creating an Amazon Linux OS based EC2 machine. Do ensure to use atleast "t2.medium" machine with 30GB of disk space. Once the machine is running, we will SSH into the machine and install Airbyte using docker compose. Follow the Airbyte [documentation page](https://docs.airbyte.com/deploying-airbyte/on-aws-ec2) on AWS EC2 installation for getting the Airbyte running on EC2.
+
+Once the Airbyte is running using docker compose, you can go to the Security Group of your EC2 machine, and edit the inbound rules to open port `8000` across all hosts using `0.0.0.0/0`. You will now be able to access the Airbyte cluster using your EC2 machine's public IP address on port 8000. If you followed the instructions as is, `airbyte` and `password` would be your username and password respectively to login into the Airbyte UI.
+
+### Create RDS MySQL database ###
+
+Now, we will be creating the RDS MySQL instance. Go to the AWS RDS service, and create a new database. You can use the `Standard Create` option, choose `MySQL` engine and use any MySQL 8 version. You can choose the deployment options as per your preference. Then, you can proceed to provide an appropriate DB cluster name, master username and password. Do ensure to select `Yes` for public access, as this helps us to easily connect to the database from any machine (we are not discussing the networking details in this blog that would be required in case the public access is disabled). The database will still be protected using the username and password. With all the settings in place, click on the button "Create Database". The database will be provisioned in a few minutes.
+
+Once the database is provisioned, you can use the RDS instance's public endpoint, port, username and password to connect to the database. Let us connect to this database using any MySQL client like MySQL Workbench or DB Visualizer from your local machine. Once connected to the database, we will create a `test` database, an `employees` table in that database, and insert three rows into the table using the following commands:
```

Comment:
```suggestion
Now, we will be creating the RDS MySQL instance. Go to the AWS RDS service, and create a new database. You can use the `Standard Create` option, choose `MySQL` engine and use any MySQL 8 version. You can choose the deployment options as per your preference. Then, you can proceed to provide an appropriate DB cluster name, master username and password.  Select `Yes` for public access, as this helps us to easily connect to the database from any machine (we are not discussing the networking details in this blog that would be required in case the public access is disabled). The database will still be protected using the username and password. With all the settings in place, click on the button "Create Database". The database will be provisioned in a few minutes.

Once the database is provisioned, you can use the RDS instance's public endpoint, port, username and password to connect to the database. Let us connect to this database using any MySQL client like MySQL Workbench or DB Visualizer from your local machine. Once connected to the database, we will create a `test` database, an `employees` table in that database, and insert three rows into the table using the following commands:
```